src/rt/lifetime.d

```diff
@@ -254,13 +303,18 @@ bool __setArrayAllocLength(ref BlkInfo info, size_t newlength, bool isshared, si
             // setting the initial length, no lock needed
             *length = cast(ubyte)newlength;
         }
+        if (typeInfoSize)
+        {
+            auto typeInfo = cast(TypeInfo*)(info.base + info.size - size_t.sizeof);
+            *typeInfo = cast() tinext;
+        }
     }
     else if(info.size < PAGESIZE)
     {
-        if(newlength + MEDPAD > info.size)
+        if(newlength + MEDPAD + typeInfoSize > info.size)
```

Comment:
Unbelievable, who wrote that code? It's the same as above only with MEDPAD and ushort instead of SMALLPAD and ubyte.